src/tiledimage.js

```diff
@@ -1656,16 +1656,22 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
      */
     _setTileLoaded: function(tile, data, cutoff, tileRequest) {
         var increment = 0,
+            completed = false,
+            eventFinished = false,
             _this = this;
 
         function getCompletionCallback() {
+            if (eventFinished) {
+                $.console.error("Event 'tile-loaded' argument getCompletionCallback must not be called asynchronously.");
```

Comment:
This is a good start, but I think it could use a little more explanation. getCompletionCallback needs to be called synchronously, but the function that it returns can be called asynchronously. Probably worth mentioning that second aspect to avoid confusion.